src/fu-engine.c

```diff
@@ -2737,6 +2737,7 @@ fu_engine_device_check_power(FuEngine *self,
 
 	/* not enough just in case */
 	if (!fu_device_has_private_flag(device, FU_DEVICE_PRIVATE_FLAG_IGNORE_SYSTEM_POWER) &&
+	    !fu_device_has_flag(device, FWUPD_DEVICE_FLAG_EMULATED) &&
```

Comment:
I guess for _now_ this is OK, but realistically I think we _should_ be using umockdev in our tests to mock upower and indicate battery or not battery to have proper test coverage.

Reply:
Ohh, do you think mocking the udev interface is a good idea? I was going to add something that just futzed with the context instead.

Reply:
I guess either way is fine, but I generally lean on less special code to say "I'm in emulation mode" means better coverage of all cases.